src/test/java/spoon/test/constructor/ConstructorTest.java

```diff
@@ -158,4 +163,42 @@ private void assertIntersectionTypeInConstructor(CtTypeReference<?> boundingType
 		assertEquals("Serializable", bounds.get(1).getSimpleName());
 		assertEquals(1, bounds.get(1).getAnnotations().size());
 	}
+
+	@Test
+	public void test_addParameterAt_addsParameterToSpecifiedPosition() {
+		Factory factory = new Launcher().getFactory();
+
+		CtConstructor<?> constructor = factory.createConstructor();
+
+		CtParameter<?> first = factory.createParameter();
+		CtTypeReference<String> firstType = factory.Type().stringType();
+		first.setSimpleName("x");
+		first.setType((CtTypeReference) firstType);
```

Comment:
Passing the argument as is (without type casting) was throwing a compilation error.

```sh
incompatible types: spoon.reflect.reference.CtTypeReference<java.lang
.String> cannot be converted to spoon.reflect.reference.CtTypeReference<capture#1 of ?>
```

I was getting confused with the types as it wasn't even allowing it to type cast to `CtTypeReference<String>` so I just used a raw type for now. Need help resolving this :)

Reply:
The type parameters must match between `CtParameter` and CtTypeReference`.

```java
CtParameter<String> first = factory.createParameter();
CtTypeReference<String> firstType = factory.Type().stringType();
first.setSimpleName("y");
first.setType(secondType);
```

The reason you can't cast to `CtTypeReference<String>` is that casting is a runtime thing, and the type parameter of a generic type is erased at compile time. That is to say, the JVM does not know the parameterization of a generic type. It's the same reason you can't do stuff like `list instanceof List<String>`. You can read [about type erasure here](https://docs.oracle.com/javase/tutorial/java/generics/erasure.html). It's quite fascinating, and massively annoying.

Reply:
> You can read about type erasure here. It's quite fascinating, and massively annoying.

😂. I shall read about it. I remember you mentioned this concept in one of our meetings.

By the way, can you run the workflow for this PR? I will be able to resolve if anything breaks.